syntax = "proto3";

option objc_class_prefix = "MMA";

package proj4;

// The greeting service definition.
service MMA {
  // Sends a greeting
  rpc ReadPage (ReadPageRequest) returns (ReadPageResponse) {}
  rpc WritePage (WritePageRequest) returns (WritePageResponse) {}
  rpc Allocate (AllocateRequest) returns (AllocateResponse) {}
  rpc Free (FreeRequest) returns (FreeResponse) {}
}

// The request message containing the user's name.
message ReadPageRequest {
  int32 arr_id = 1;
  int32 vid = 2;
  int32 offset = 3;
}

message ReadPageResponse {
  bool ok = 1;
  int32 data = 2;
}

message WritePageRequest {
  int32 arr_id = 1;
  int32 vid = 2;
  int32 offset = 3;
  int32 value = 4;
}

message WritePageResponse {
}

message AllocateRequest {
  int32 size = 1;
}

message AllocateResponse {
  int32 arr_id = 2;
}

message FreeRequest {
  int32 arr_id = 1;
}

message FreeResponse {
}
